runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/EngineRule.java

```diff
@@ -221,14 +213,14 @@ public Statement apply(
                 if (configurationRoot != null)
                 {
                     String resourceName = String.format("%s/%s", configurationRoot, config.value());
-
-                    configURL = testClass.getClassLoader().getResource(resourceName);
+                    String configURL = requireNonNull(testClass.getClassLoader().getResource(resourceName)).toString();
+                    configure(ENGINE_CONFIG_URL.name(), configURL);
```

Comment:
There is an overload of `configure` that takes parameter types `PropertyDef<T>, T`.

```
URL configURL = testClass.getClassLoader().getResource(resourceName));
configure(ENGINE_CONFIG_URL, configURL);
```

Please move `requireNotNull` checks to `configure(..., value)` overloads instead of making the call site more complex.